### PR TITLE
Add Snowflake DAG Example

### PR DESCRIPTION
**Pull request by denimalpaca**
This pipeline is very similar to the Bigquery and Redshift EL 3 DAGs in structure and usage. It serves as an example to show how data quality can be added to a Snowflake DAG. It uses two operators, SnowflakeCheckOperator and SnowflakeValueCheckOperator, that do not currently exist in the Snowflake provider; a PR has been opened in Airflow to add these (https://github.com/apache/airflow/pull/17741).

### CODE REVIEW

File: dags/snowflake_examples/simple_snowflake_el.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
+from airflow import DAG, AirflowException
```
Comment (josh-fell) [inline dags/snowflake_examples/simple_snowflake_el.py]: Doesn't look like `AirflowException` nor `Variable` modules are used in this DAG. Can you remove these imports?

File: dags/snowflake_examples/simple_snowflake_el.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+SNOWFLAKE_SCHEMA = 'SANDBOX_BENJI'
+SNOWFLAKE_WAREHOUSE = 'DEMO'
+SNOWFLAKE_DATABASE = 'DWH_LEGACY'
+SNOWFLAKE_ROLE = 'BENJI'
+SNOWFLAKE_FORESTFIRE_TABLE = 'forestfires'
+
+# These args will get passed on to each operator
+# You can override them on a per-task basis during operator initialization
+default_args = {
```
Comment (josh-fell) [inline dags/snowflake_examples/simple_snowflake_el.py]: Can you update the `default_args` and `start_date` pattern to what we discussed in the Airflow/dbt guide comments?

File: dags/snowflake_examples/simple_snowflake_el.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        database=SNOWFLAKE_DATABASE,
+        schema=SNOWFLAKE_SCHEMA,
+        role=SNOWFLAKE_ROLE,
+        params={"table_name": SNOWFLAKE_FORESTFIRE_TABLE}
+    )
+
+    begin = DummyOperator(task_id='begin')
+    end = DummyOperator(task_id='end')
+
+    begin >> create_table >> load_data
```
Comment (josh-fell) [inline dags/snowflake_examples/simple_snowflake_el.py]: Could these dependencies be simplified using `airflow.models.baseoperator.chain()`? Something like:

```
chain(begin, create_table, load_data, [quality_check_group, check_bq_row_count], delete_table, end)

**Review by josh-fell — CHANGES_REQUESTED** [review]

File: dags/snowflake_examples/simple_snowflake_el.py (hunk 3 above)
Comment (denimalpaca) [reply dags/snowflake_examples/simple_snowflake_el.py]: I didn't even know about `chain()`! Thanks for letting me know.

**Review by josh-fell — APPROVED** [review]

**Pull request merged**